microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -132,6 +132,46 @@ def verify_dpdk_ovs(
         finally:
             ovs.stop_ovs()
 
+    @TestCaseMetadata(
+        description="""
+           Build and run DPDK multiprocess client/server sample application
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            min_nic_count=2,
+            network_interface=Sriov(),
+        ),
+    )
+    def verify_dpdk_multiprocess(
+        self, node: Node, log: Logger, variables: Dict[str, Any]
+    ) -> None:
+        pmd = "failsafe"
+        # initialize DPDK with sample applications selected for build
+        test_kit = initialize_node_resources(
+            node,
+            log,
+            variables,
+            pmd,
+            sample_apps=[
+                "multi_process/client_server_mp/mp_server",
+                "multi_process/client_server_mp/mp_client",
+            ],
+        )
+
+        # enable hugepages needed for dpdk EAL
+        _init_hugepages(node)
+
+        # setup and run mp_server application
+        examples_path = test_kit.testpmd.dpdk_build_path.joinpath("examples")
+        server_app_path = examples_path.joinpath("dpdk-mp_server")
+        kit_cmd_pair = {test_kit: f"{server_app_path} --help"}
+        output: Dict[DpdkTestResources, str] = dict()
+        task_manager = _start_testpmd_concurrent(kit_cmd_pair, 5, log, output)
+
+        # TODO: Just enough to test the changes, rest of test incoming.
+
+        task_manager.wait_for_all_workers()
```

Comment:
I created a new wrapper called `run_in_parallel`. It works well for the simple concurrent tasks. Check code for examples.

Reply:
I need to do stuff while concurrent things are running usually. Maybe I could extend run_in_parallel and add an async version?

Reply:
If so, you can start them like run_async one by one, and then verify. Finally, just wait them finished or kill them. The `run_in_parallel` is to block until all done.

Reply:
That's sort of what my current wrapper is doing, though I already put in some work to make the parallel util useful for me without changing how run_in_parallel works for other callers. I'll add the commit here and you can see what I mean.

Reply:
testing it out rn to make sure everything works still :)

Reply:
I would prefer a fancier `collect_result` function and would like the wrapper to not busy wait for the results, so this gives me what I need. Otherwise I can just keep using the wrapper I had before, though I like the new version though now that I've written it 😂

Reply:
ok, please add the async version. Please also add comments to explain it runs all methods at the same time.